gammapy/spectrum/fit.py

```diff
@@ -204,8 +204,11 @@ def _predict_counts_helper(self, obs, model, forward_folded=True):
                                               e_reco=binning)
             counts = temp.data.data
         else:
-            # TODO: This could also be part of calculate predicted counts
-            counts = model.integral(binning[:-1], binning[1:])
+            temp = calculate_predicted_counts(model=model,
```

Comment:
``model.integral`` returns whatever the unit of the model amplitude is (times energy). Do you want to update the docstring of ``SpectrumFit`` in this PR, or should I do this in a separate one?